packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -793,16 +798,21 @@ class TextSelectionOverlay {
     }
 
     final TextSelection newSelection;
+    final bool dragStartSelectionNormalized = _dragStartSelection!.extentOffset >= _dragStartSelection!.baseOffset;
     switch (defaultTargetPlatform) {
       // On Apple platforms, dragging the base handle makes it the extent.
       case TargetPlatform.iOS:
       case TargetPlatform.macOS:
-        newSelection = TextSelection(
-          extentOffset: position.offset,
-          baseOffset: _selection.start,
-        );
-        if (position.offset <= _selection.start) {
-          return; // Don't allow order swapping.
+        if (dragStartSelectionNormalized) {
```

Comment:
Nit: You could also remove this if/else and just do a ternary for the baseOffset. Either way is fine though.

Reply:
Done.